components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1763,7 +1759,8 @@ private Set<org.wso2.carbon.user.core.common.User> filterUsers(Node node, int of
             throws CharonException, BadRequestException {
 
         // Filter users when filter by group.
-        if (SCIMConstants.UserSchemaConstants.GROUP_URI.equals(((ExpressionNode) node).getAttributeValue())) {
+        if (!SCIMCommonUtils.isGroupBasedUserFilteringImprovementsEnabled() &&
+                SCIMConstants.UserSchemaConstants.GROUP_URI.equals(((ExpressionNode) node).getAttributeValue())) {
```

Comment:
Fix the 4th problem mentioned here https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/609#issuecomment-2644715275

We have provided a bypass for filter single attribute of group. It lacks support for pagination and other features (This by pass is used for filter roles from users endpoint, but it is blocked in the master by the given config.)

Hence its better to let groups filter to flow through the below standard filtering logic which works correctly.